### PR TITLE
DOC-2255: Fix description of flowThreadPoolSize in Corda Configuration Field pages

### PR DESCRIPTION
**Pull request by AaronForinton**
In configuration fields tuning section, two statements regarding the default value of flowThreadPoolSize appears contradictory:

1. The default value is 2 times the number of cores available (the minimum is 30 if there are less than 15 cores) which was found to be working efficiently in performance testing.
2. For these reasons, the default size for the flow framework thread pool is the minimum between two times the available number of processors and 30.

2 is correct as confirmed by engineer. Eliminating 1.

### CODE REVIEW

**Review by bpaunescu — APPROVED** [review]
The corrected statements about the default value of the `flowThreadPoolSize`  look OK.

File: content/en/platform/corda/4.3/enterprise/corda-configuration-file.md [hunk 1]
```diff
@@ -464,27 +464,21 @@ To add additional data source properties (for a specific JDBC driver) use the `d
...
+  Otherwise, a currently active flow must be finished or suspended before the suspended flow can be woken
+  up to handle the event. This can have serious performance implications if the flow thread pool is too small,
+  as a flow cannot be suspended while in a database transaction, or without checkpointing its state first.
+  Corda Enterprise allows the node operators to configure the number of threads the state machine manager can use to execute flows in
+  parallel, allowing more than one flow to be active and/or use resources at the same time.
+  The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
+using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.3/enterprise/corda-configuration-file.md:479]:
```suggestion
using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.4/enterprise/node/setup/corda-configuration-file.md [hunk 2]
```diff
@@ -480,17 +480,13 @@ This is an optimisation for sharing transaction backchains. Corda Enterprise nod
...
-      so for n threads, the database system must have at least n+1 connections available. Also, the database
-      must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
-      using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
-      architecture. For these reasons, the default size for the flow framework thread pool is the minimum between two times the available number of processors and 30. Overriding this value in the configuration allows to specify any number.
-
-    `rpcThreadPoolSize`
+The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
+using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.4/enterprise/node/setup/corda-configuration-file.md:485]:
```suggestion
using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.4/enterprise/node/setup/corda-configuration-file.md [hunk 3]
```diff
@@ -499,16 +495,16 @@ This is an optimisation for sharing transaction backchains. Corda Enterprise nod
       On a multicore machine with a large ``flowThreadPoolSize``, this might need to be increased, to avoid flow
       threads being idle while the payload is being deserialized and the flow invocation run.
 
-      If there are idling flow threads while RPC calls are queued, it might be worthwhile increasing this number slightly.
+If there are idling flow threads while RPC calls are queued, it might be worthwhile increasing this number slightly.
       Valid values for this property are between 4 (that is the number used for the single threaded state machine in
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.4/enterprise/node/setup/corda-configuration-file.md:499]:
```suggestion
      Valid values for this property are between four (that is the number used for the single threaded state machine in
```

File: content/en/platform/corda/4.5/enterprise/operations/deployment/corda-configuration-fields.md [hunk 4]
```diff
@@ -320,18 +320,14 @@ A larger number of flows can be suspended, e.g. waiting for reply from a counter
...
 
-The default value is 2 times the number of cores available (the minimum is 30 if there are less than 15 cores) which was found to be working efficiently in performance testing.
-
-The ideal value for this parameter depends on a number of factors.
-
-The main ones are the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
 must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
 using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.5/enterprise/operations/deployment/corda-configuration-fields.md:329]:
```suggestion
using for example, H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.6/enterprise/node/setup/corda-configuration-fields.md [hunk 5]
```diff
@@ -330,24 +330,20 @@ Tuning is a section within the Corda Node configuration file that contains perfo
...
-    must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
-    using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
-    architecture. For these reasons, the default size for the flow framework thread pool is the minimum between two times the available number of processors and 30. Overriding this value in the configuration allows to specify any number.
+  The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+  must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
+  using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.6/enterprise/node/setup/corda-configuration-fields.md:345]:
```suggestion
  using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.7/enterprise/node/setup/corda-configuration-fields.md [hunk 6]
```diff
@@ -326,30 +326,26 @@ Tuning is a section within the Corda Node configuration file that contains perfo
...
-    must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
-    using e.g., H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
-    architecture. For these reasons, the default size for the flow framework thread pool is the minimum between two times the available number of processors and 30. Overriding this value in the configuration allows to specify any number.
+  The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+  must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
+  using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.7/enterprise/node/setup/corda-configuration-fields.md:347]:
```suggestion
  using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.8/enterprise/node/setup/corda-configuration-fields.md [hunk 7]
```diff
@@ -323,34 +323,30 @@ Allows fine-grained controls of various features only available in the enterpris
...
-    must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
-    using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
-    architecture. For these reasons, the default size for the flow framework thread pool is the minimum between two times the available number of processors and 30. Overriding this value in the configuration allows to specify any number.
+  The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+  must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
+  using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.8/enterprise/node/setup/corda-configuration-fields.md:348]:
```suggestion
  using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.8/enterprise/node/setup/corda-configuration-fields.md [hunk 8]
```diff
@@ -366,24 +362,24 @@ Tuning is a section within the Corda Node configuration file that contains perfo
...
+- The size of the in-memory Artemis buffer for messages, in bytes. Note that there is a lower bound to the buffer size, which is calculated based on the maximum message size of the network parameters to ensure messages of any allowed size can be stored successfully. As a result, any value lower than this bound will be ignored with the appropriate logging. This bound is also used as the default, if no value is specified.
 
 - `extraNetworkMapKeys`
 
     An optional list of private network map UUIDs. Your node will fetch the public network and private network maps based on these keys.
     Private network UUID should be provided by network operator and lets you see nodes not visible on public network.
 
-    **Important: This is a temporary feature for onboarding network participants that limits their visibility for privacy reasons.**
+    **Important: This is a temporary feature for on-boarding network participants that limits their visibility for privacy reasons.**
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.8/enterprise/node/setup/corda-configuration-fields.md:374]:
```suggestion
{{< note >}}

This is a temporary feature for on-boarding network participants that limits their visibility for privacy reasons.

{{< /note >}}
```

File: content/en/platform/corda/4.2/enterprise/corda-configuration-file.md [hunk 9]
```diff
@@ -119,16 +119,17 @@ Mode used when setting up the Artemis client. Supported modes are: DEFAULT (5 in
 FAIL_FAST (no initial attempts, no reconnect attempts), CONTINUOUS_RETRY (infinite initial and reconnect attempts, starting retry interval of 5 seconds with an exponential back-of multiplier of 1.5 up for up to 5 minutes retry interval).*Default:* DEFAULTList of Artemis Server back-up addresses. If any back-ups are specified, the client will be configured to automatically failover to the first server it can connect to.*Default:* empty list
 
 
-Performance tuning parameters for Corda EnterpriseThe number of threads available to handle flows in parallel. This is the number of flows
+## Performance tuning parameters for Corda Enterprise
+
+The number of threads available to handle flows in parallel. This is the number of flows
 that can run in parallel doing something and/or holding resources like database connections.
 A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.2/enterprise/corda-configuration-file.md:126]:
```suggestion
A larger number of flows can be suspended, for example, waiting for reply from a counterparty.
```

File: content/en/platform/corda/4.3/enterprise/corda-configuration-file.md [hunk 10]
```diff
@@ -464,27 +464,21 @@ To add additional data source properties (for a specific JDBC driver) use the `d
 Performance tuning parameters for Corda Enterprise
 
 * `flowThreadPoolSize`
+  
   The number of threads available to handle flows in parallel. This is the number of flows
-  that can run in parallel doing something and/or holding resources like database   connections.
-
+  that can run in parallel doing something and/or holding resources like database connections.
   A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.3/enterprise/corda-configuration-file.md:470]:
```suggestion
  A larger number of flows can be suspended, for example, waiting for reply from a counterparty.
```

File: content/en/platform/corda/4.5/enterprise/node/setup/corda-configuration-fields.md [hunk 11]
```diff
@@ -313,24 +313,20 @@ Tuning is a section within the Corda Node configuration file that contains perfo
 
 - `flowThreadPoolSize`
 
-    The number of threads available to handle flows in parallel. This is the number of flows
-    that can run in parallel doing something and/or holding resources like database connections.
-    A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
-    When a response arrives, a suspended flow will be woken up if there are any available threads in the thread pool.
+  The number of threads available to handle flows in parallel. This is the number of flows
+  that can run in parallel doing something and/or holding resources like database connections.
+  A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.5/enterprise/node/setup/corda-configuration-fields.md:318]:
```suggestion
  A larger number of flows can be suspended, for example, waiting for reply from a counterparty.
```

File: content/en/platform/corda/4.6/enterprise/node/setup/corda-configuration-fields.md [hunk 12]
```diff
@@ -330,24 +330,20 @@ Tuning is a section within the Corda Node configuration file that contains perfo
 
 - `flowThreadPoolSize`
 
-    The number of threads available to handle flows in parallel. This is the number of flows
-    that can run in parallel doing something and/or holding resources like database connections.
-    A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
-    When a response arrives, a suspended flow will be woken up if there are any available threads in the thread pool.
+  The number of threads available to handle flows in parallel. This is the number of flows
+  that can run in parallel doing something and/or holding resources like database connections.
+  A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.6/enterprise/node/setup/corda-configuration-fields.md:335]:
```suggestion
  A larger number of flows can be suspended, for example, waiting for reply from a counterparty.
```

File: content/en/platform/corda/4.6/enterprise/operations/deployment/corda-configuration-fields.md [hunk 13]
```diff
@@ -337,18 +337,14 @@ A larger number of flows can be suspended, e.g. waiting for reply from a counter
...
 
-The default value is 2 times the number of cores available (the minimum is 30 if there are less than 15 cores) which was found to be working efficiently in performance testing.
-
-The ideal value for this parameter depends on a number of factors.
-
-The main ones are the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
 must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
 using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.6/enterprise/operations/deployment/corda-configuration-fields.md:346]:
```suggestion
using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.7/enterprise/node/setup/corda-configuration-fields.md [hunk 14]
```diff
@@ -326,30 +326,26 @@ Tuning is a section within the Corda Node configuration file that contains perfo
...
 
 - `flowThreadPoolSize`
 
-    The number of threads available to handle flows in parallel. This is the number of flows
-    that can run in parallel doing something and/or holding resources like database connections.
-    A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
-    When a response arrives, a suspended flow will be woken up if there are any available threads in the thread pool.
+  The number of threads available to handle flows in parallel. This is the number of flows
+  that can run in parallel doing something and/or holding resources like database connections.
+  A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.7/enterprise/node/setup/corda-configuration-fields.md:337]:
```suggestion
  A larger number of flows can be suspended, for example, waiting for reply from a counterparty.
```

File: content/en/platform/corda/4.7/enterprise/operations/deployment/corda-configuration-fields.md [hunk 15]
```diff
@@ -339,18 +339,14 @@ A larger number of flows can be suspended, e.g. waiting for reply from a counter
...
 
-The default value is 2 times the number of cores available (the minimum is 30 if there are less than 15 cores) which was found to be working efficiently in performance testing.
-
-The ideal value for this parameter depends on a number of factors.
-
-The main ones are the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
 must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
 using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.7/enterprise/operations/deployment/corda-configuration-fields.md:348]:
```suggestion
using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

File: content/en/platform/corda/4.8/enterprise/node/setup/corda-configuration-fields.md [hunk 16]
```diff
@@ -323,34 +323,30 @@ Allows fine-grained controls of various features only available in the enterpris
...
 
 - `flowThreadPoolSize`
 
-    The number of threads available to handle flows in parallel. This is the number of flows
-    that can run in parallel doing something and/or holding resources like database connections.
-    A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
-    When a response arrives, a suspended flow will be woken up if there are any available threads in the thread pool.
+  The number of threads available to handle flows in parallel. This is the number of flows
+  that can run in parallel doing something and/or holding resources like database connections.
+  A larger number of flows can be suspended, e.g. waiting for reply from a counterparty.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.8/enterprise/node/setup/corda-configuration-fields.md:338]:
```suggestion
  A larger number of flows can be suspended, for example, waiting for reply from a counterparty.
```

File: content/en/platform/corda/4.8/enterprise/operations/deployment/corda-configuration-fields.md [hunk 17]
```diff
@@ -335,18 +335,14 @@ A larger number of flows can be suspended, e.g. waiting for reply from a counter
...
 
-The default value is 2 times the number of cores available (the minimum is 30 if there are less than 15 cores) which was found to be working efficiently in performance testing.
-
-The ideal value for this parameter depends on a number of factors.
-
-The main ones are the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
+The ideal value for this parameter depends on a number of factors. These include the hardware the node is running on, the performance profile of the flows, and the database instance backing the node as datastore. Every thread will open a database connection, so for n threads, the database system must have at least n+1 connections available. Also, the database
 must be able to actually cope with the level of parallelism to make the number of threads worthwhile - if
 using e.g. H2, any number beyond 8 does not add any substantial benefit due to limitations with its internal
```
Comment (AnnaK-R3) [inline content/en/platform/corda/4.8/enterprise/operations/deployment/corda-configuration-fields.md:344]:
```suggestion
using for example H2, any number beyond eight does not add any substantial benefit due to limitations with its internal
```

**Review by AnnaK-R3 — APPROVED** [review]

**Pull request merged**